code/app.py

```diff
@@ -26,29 +30,39 @@
 def static_file(path):
     return app.send_static_file(path)
 
-
-@app.route("/api/config", methods=["GET"])
+AZURE_AUTH_TYPE = os.environ.get("AZURE_AUTH_TYPE", "keys")
+# Initialize Azure keys based on authentication type and environment settings. 
+# When AZURE_AUTH_TYPE is not 'rbac' and USE_KEY_VAULT environment variable is set, 
+# Azure keys are securely fetched from Azure Key Vault using DefaultAzureCredential. 
+# Otherwise, keys are obtained from environment variables, with fallbacks to None or 
+# an empty string depending on the AZURE_AUTH_TYPE.
+if not AZURE_AUTH_TYPE == 'rbac' and os.environ.get("USE_KEY_VAULT"):
+    credential = DefaultAzureCredential()
+    secret_client = SecretClient(os.environ.get("AZURE_KEY_VAULT_ENDPOINT"), credential)
+    AZURE_SEARCH_KEY = secret_client.get_secret(os.environ.get("AZURE_SEARCH_KEY")).value
+    AZURE_OPENAI_KEY = secret_client.get_secret(os.environ.get("AZURE_OPENAI_KEY")).value
+    AZURE_SPEECH_KEY = secret_client.get_secret(os.environ.get("AZURE_SPEECH_SERVICE_KEY")).value
+else:
```

Comment:
We generally recommend avoiding Python ternaries for readability. I think it'd be clearer to do

elif AZURE_AUTH_TYPE == 'rbac':
  AZURE_SEARCH_KEY = None
  AZURE_OPENAI_KEY = ""
  AZURE_SPEECH_KEY = None
else:
  AZURE_SEARCH_KEY = os.environ.get("AZURE_SEARCH_KEY")
  AZURE_OPENAI_KEY = os.environ.get("AZURE_OPENAI_KEY")
  AZURE_SPEECH_KEY = os.environ.get("AZURE_SPEECH_SERVICE_KEY")

I'd probably put the rbac case first and then the two key cases.

Just a style thing though, not a big deal. Python ternaries are unfortunately not the most readable, the way they're ordered.